internal/namingschema/namingschema.go

```diff
@@ -23,55 +28,74 @@ const (
 	SchemaV1
 )
 
-const (
-	defaultSchemaVersion = SchemaV0
-)
+type Config struct {
+	NamingSchemaVersion           Version
+	RemoveIntegrationServiceNames bool
+	DDService                     string
+}
 
 var (
-	sv int32
-
-	useGlobalServiceName   bool
-	useGlobalServiceNameMu sync.RWMutex
+	activeNamingSchema            int32
+	removeIntegrationServiceNames bool
+	mu                            sync.RWMutex
 )
 
-// ParseVersion attempts to parse the version string.
-func ParseVersion(v string) (Version, bool) {
-	switch strings.ToLower(v) {
-	case "", "v0":
-		return SchemaV0, true
-	case "v1":
-		return SchemaV1, true
-	default:
-		return SchemaV0, false
-	}
-}
-
 // GetVersion returns the global naming schema version used for this application.
 func GetVersion() Version {
-	return Version(atomic.LoadInt32(&sv))
+	return Version(atomic.LoadInt32(&activeNamingSchema))
 }
 
-// SetVersion sets the global naming schema version used for this application.
-func SetVersion(v Version) {
-	atomic.StoreInt32(&sv, int32(v))
+// SetRemoveIntegrationServiceNames sets the value of the RemoveIntegrationServiceNames setting for this application.
+func SetRemoveIntegrationServiceNames(v bool) {
+	mu.Lock()
+	defer mu.Unlock()
+	removeIntegrationServiceNames = v
 }
 
-// SetDefaultVersion sets the default global naming schema version.
-func SetDefaultVersion() Version {
-	SetVersion(defaultSchemaVersion)
-	return defaultSchemaVersion
+func LoadFromEnv() {
+	schemaVersionStr := os.Getenv("DD_TRACE_SPAN_ATTRIBUTE_SCHEMA")
+	if v, ok := parseVersionStr(schemaVersionStr); ok {
+		setVersion(v)
+	} else {
+		setVersion(SchemaV0)
+		log.Warn("DD_TRACE_SPAN_ATTRIBUTE_SCHEMA=%s is not a valid value, setting to default of v%d", schemaVersionStr, v)
+	}
+	// Allow DD_TRACE_SPAN_ATTRIBUTE_SCHEMA=v0 users to disable default integration (contrib AKA v0) service names.
+	// These default service names are always disabled for v1 onwards.
+	removeIntegrationServiceNames = internal.BoolEnv("DD_TRACE_REMOVE_INTEGRATION_SERVICE_NAMES_ENABLED", false)
```

Comment:
Shouldn't this be also protected with a mutext? `ReloadConfig` may cause race conditions.